plugins/woocommerce/includes/admin/meta-boxes/class-wc-meta-box-coupon-data.php

```diff
@@ -388,6 +388,11 @@ public static function save( $post_id, $post ) {
 				'email_restrictions'          => array_filter( array_map( 'trim', explode( ',', wc_clean( $_POST['customer_email'] ) ) ) ),
 			)
 		);
+
+		if ( is_wp_error( $errors ) ) {
+			WC_Admin_Meta_Boxes::add_error( $errors->get_error_message() );
+		}
```

Comment:
If there are multiple errors, only one is displayed
- Enter negative coupon amount
- Enter maximum spend < minimum

With the suggestion, all errors will be displayed

```suggestion
		if ( is_wp_error( $errors ) ) {
			foreach ( $errors->get_error_messages() as $error_message ) {
				WC_Admin_Meta_Boxes::add_error( $error_message );
			}
		}
```

![image](https://github.com/user-attachments/assets/1bf03f1b-0b6b-4105-89a6-b70efac4dd66)

Reply:
Hey @PanosSynetos, 

I agree that this approach is better, as users have more context about the fields that were not saved. 

The reason why I only displayed 1 error was to be consistent with how product errors are displayed: [https://github.com/woocommerce/woocommerce/blob/e68ccf8620455a1ab5268ca7bd7b5ce2f2[…]ce/includes/admin/meta-boxes/class-wc-meta-box-product-data.php](https://github.com/woocommerce/woocommerce/blob/e68ccf8620455a1ab5268ca7bd7b5ce2f2190845/plugins/woocommerce/includes/admin/meta-boxes/class-wc-meta-box-product-data.php#L423-L425), but in that case it is quite difficult to post a request with more than 1 invalid options from the UI -- some DOM editing is required to post invalid values in some fields.